### PR TITLE
Feature/cvsb 7675

### PR DESCRIPTION
**Pull request by m-ujaffer**

### CODE REVIEW

File: src/assets/Enums.ts [hunk 1]
```diff
@@ -56,3 +56,9 @@ export const TEST_STATUS = {
     SUBMITTED: "submitted",
     CANCELLED: "cancelled"
 };
+
+export const HGV_TRL_ROADWORTHINESS_TEST_TYPES = {
+    id_hgv_trl : "122",
```
Comment (KHodderDVSA) [inline src/assets/Enums.ts]: conventionally all caps for constants

File: src/services/TestResultsService.ts [hunk 2]
```diff
@@ -491,10 +491,18 @@ export class TestResultsService {
...
+  public isHGVTRLRoadworthinessTest(testTypeId: string): boolean {
+    return Object.values(HGV_TRL_ROADWORTHINESS_TEST_TYPES).includes(testTypeId);
+   }
+
   public generateCertificateNumber(payload: ITestResultPayload) {
     if (payload.testStatus === TEST_STATUS.SUBMITTED) {
       payload.testTypes.forEach((testType) => {
-        if (testType.testTypeClassification === TEST_TYPE_CLASSIFICATION.ANNUAL_WITH_CERTIFICATE && testType.testResult !== TEST_RESULT.ABANDONED && !this.isTestTypeAdr(testType) && !this.isTestTypeLec(testType)) {
+        // CVSB-7675 if vehicle type is HGV/TRL and testTypeId is Roadworthiness test and testResult is pass then testNumber = certificateNumber
+       if ( (payload.vehicleType === VEHICLE_TYPES.HGV || payload.vehicleType === VEHICLE_TYPES.TRL && this.isHGVTRLRoadworthinessTest(testType.testTypeId) && testType.testResult === TEST_RESULT.PASS)
```
Comment (KHodderDVSA) [inline src/services/TestResultsService.ts]: Getting extremely unwieldy. I'd be inclined to split this whole thing out to a Util function with some more named functions, just to make the logic easier to read e.g. isHGVorTRLVehicle(payload),

File: tests/unit/insertTestResult.unitTest.ts [hunk 3]
```diff
@@ -1500,4 +1500,80 @@ describe("insertTestResult", () => {
                 });
         });
     });
+
+    context("when inserting a testResult that is a vehicleType of hgv or trl and it contains at least one Roadworthiness test type and the test result on the Roadworthiness test type is pass", () => {
+        it("then a testNumber is generated and the inserted test result should set the testNumber as the certificateNumber.", () => {
```
Comment (KHodderDVSA) [inline tests/unit/insertTestResult.unitTest.ts]: This whole test is easier to achieve if you just test generateCertificateNumber. It's private but can get around that either with @ts-ignore, or (testResultsService as any).generateCertificateNumber

File: src/assets/Enums.ts (hunk 1 above)
Comment (oalexandru) [reply src/assets/Enums.ts]: I would have used an array(as I think there are more roadworthiness test-types)

File: tests/unit/generateCertificateNumber.unitTest.ts [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+import {cloneDeep} from "lodash";
+
+describe("TestResultsService calling generateExpiryDate", () => {
+    let testResultsService: TestResultsService | any;
+    let MockTestResultsDAO: jest.Mock;
+    let testResultsMockDB: any;
+    let testResultsPostMock: any;
+
+    beforeEach(() => {
+        testResultsMockDB = JSON.parse(fs.readFileSync(path.resolve(__dirname, "../resources/test-results.json"), "utf8"));
```
Comment (KHodderDVSA) [inline tests/unit/generateCertificateNumber.unitTest.ts]: could you not use imports for these? Should already have required tsconfig change.

**Pull request merged**